### PR TITLE
Add outputs to the exec provider and exec actions

### PR DESCRIPTION
**Pull request by Orzelius**
closes https://github.com/garden-io/garden/issues/3998
closes https://github.com/garden-io/garden/issues/4437

juicy details in commits

### CODE REVIEW

**Comment by Orzelius** [comment]
This is ready for review I just need some help figuring out how to get rid of the nvm message from the project `initScript` log https://app.circleci.com/pipelines/github/garden-io/garden/18056/workflows/94b28f1e-454e-497d-980d-5735b078f6a5/jobs/315550?invite=true#step-113-532

**Comment by Orzelius** [comment]
marking as wip until I figure out the tests

**Comment by Orzelius** [comment]
Ready for review

File: .circleci/config.yml [hunk 1]
```diff
@@ -753,17 +726,14 @@ jobs:
             GARDEN_SKIP_TESTS="<<parameters.skipTests>>" yarn _integ
       - run:
           name: Deploy demo-project
-          command: sudo -E ./bin/garden deploy --root examples/demo-project
```
Comment (Orzelius) [inline .circleci/config.yml:756]: removing this will probably reduce all kinds of odd behavior

**Review by Walther — DISMISSED** [review]
good work, thanks! 🎉

**Review by Walther — APPROVED** [review]

**Pull request merged**